src/decorators/inject.ts

```diff
@@ -0,0 +1,62 @@
+import WeakMap from '@dojo/shim/WeakMap';
+import { handleDecorator, WidgetBase } from './../WidgetBase';
+import { beforeProperties } from './beforeProperties';
+import { RegistryLabel } from './../interfaces';
+
+/**
+ * Map of instances against registered injectors.
+ */
+const registeredInjectorsMap: WeakMap<WidgetBase, any[]> = new WeakMap();
+
+/**
+ * Defines the contract requires for the get properties function
+ * used to map the injected properties.
+ */
+export interface GetProperties<T = any> {
+	(context: any, properties: T): T;
```

Comment:
is this `context`?

Reply:
this is the `context` from the `Injector`

Reply:
what is a `context`? can we get a better name?

Reply:
I am receptive to better names.

Reply:
I can't participate in suggesting a name without knowing what it is 😄

Reply:
It's anything that you want to inject, it is what gets passed to the `getProperties` function that returns the values that get injected. It was referred to as a context in the previous implementation also.

```ts
const injector = new Injector(myThingToInject);

const myThingToInject = injector.get();
```

Reply:
injecting a `payload`?

Reply:
If you want it to be payload, I don't have any strong attachment to `context` or objections to `payload` - they both work.

Reply:
the reason i don't like `context` is the baggage that comes from react with it

Reply:
Yup I understand that. Let me update that for you - I think both are okay and if using `context` could lead to confusing that is it enough reason to change.

Reply:
`value`?  Since you are injecting a _value_ of some sort that can be `any`.  Do we actually expect it to be `any` or is it an `object`?

Reply:
Aye it can be a `any`, just as the `payload` of an `Injector` can be.